### PR TITLE
[jaeger-operator] Updated jaeger-operator to version 1.21

### PR DESCRIPTION
**Pull request by chgl**
#### What this PR does

Updates the jaeger-operator image to v1.21.2.

#### Checklist

- [x] [DCO](https://github.com/jaegertracing/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Commits are [GPG signed](https://docs.github.com/en/github/authenticating-to-github/about-commit-signature-verification)
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (`[jaeger]` or `[jaeger-operator]`)

### CODE REVIEW

File: charts/jaeger/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v1
 appVersion: 1.20.0
 description: A Jaeger Helm chart for Kubernetes
 name: jaeger
-version: 0.39.7
+version: 0.40.0
```
Comment (mehta-ankit) [inline charts/jaeger/Chart.yaml:5]: FYI, Jaeger chart has already been updated to 1.21.0 version of jaeger
https://github.com/jaegertracing/helm-charts/blob/master/charts/jaeger/Chart.yaml

Comment (mehta-ankit) [reply charts/jaeger/Chart.yaml:5]: You can just update the operator if you want on this PR.

**Review by mehta-ankit — CHANGES_REQUESTED** [review]
requested change to update the PR to just update operator and not jaeger helm chart (as its already up to date)

File: charts/jaeger/Chart.yaml (hunk 1 above)
Comment (chgl) [reply charts/jaeger/Chart.yaml]: Sounds good!I just rebased and changed the PR title.

**Review by mehta-ankit — APPROVED** [review]
LGTM

**Pull request merged**